### ISSUE TITLE
multi-band onset strength

### ISSUE THREAD

**Issue by bmcfee** [bmcfee/librosa#226]
#221 will require multi-band onset strength calculation.  Since the `onset_strength` is quasi-feature-agnostic, the interface for making this happen isn't 100% obvious.

I think this can be done with something like the following:

```python
>>> channels = np.arange(0, 128, 16)
>>> odf_multi = onset_strength_multi(channels, y=y, sr=sr, n_mels=128, **kwargs)
```

where `channels` specifies the (left) edges of the target rows in a fashion similar to `feature.sync`.  `odf_multi` will then duplicate [this region](https://github.com/bmcfee/librosa/blob/master/librosa/onset.py#L249-L261) of `onset_strength` to compute a spectrogram `S`, and construct `odf_multi` by passing horizontal slices `S[channels[i]:channels[i+1]]` into `onset_strength`.

The output `odf_multi` will then have shape `(len(channels), t)` and each row corresponds to an onset strength envelope for the specified band.

If `channels` falls outside the shape of the spectrogram, an exception is raised.

**Comment by bmcfee** [comment]
re: #228 , `channels` should also support a list-like of `slice` objects.

**Comment by bmcfee** [comment]
An alternative implementation of this would simply add ``channels`` as a parameter to `onset_strength`.  This might end up being more efficient, but would require an API change since the default output shape would be `(1, t)` instead of `(t,)`.

**Comment by bmcfee** [comment]
> As a compromise, we could migrate the core of onset_strength into onset_strength_multiband, and then turn onset_strength into a wrapper that provides a single stub slice and flattens the output for backwards compatibility.

This is now implemented in #229.

**Comment by bmcfee** [comment]
resolved by merging #229
